File: java/exercises/src/test/java/io/github/ajoz/workshop/fp/tests/theory/BetweenSupplier.java
package io.github.ajoz.workshop.fp.tests.theory;

import org.junit.experimental.theories.ParameterSignature;
import org.junit.experimental.theories.ParameterSupplier;
import org.junit.experimental.theories.PotentialAssignment;

import java.util.ArrayList;
import java.util.List;

public final class BetweenSupplier extends ParameterSupplier {
    @Override
    public List<PotentialAssignment> getValueSources(final ParameterSignature sig) throws Throwable {
        final List<PotentialAssignment> list = new ArrayList<>();
        final Between annotation = sig.getAnnotation(Between.class);

        for (int i = annotation.first(); i <= annotation.last(); i++)
            list.add(PotentialAssignment.forValue("ints", i));
        return list;
    }
}

File: kotlin/tools/src/test/kotlin/io/github/ajoz/workshop/fp/kotlin/tests/theory/BetweenSupplier.kt
package io.github.ajoz.workshop.fp.kotlin.tests.theory

import org.junit.experimental.theories.ParameterSignature
import org.junit.experimental.theories.ParameterSupplier
import org.junit.experimental.theories.PotentialAssignment

class BetweenSupplier : ParameterSupplier() {
    @Throws(Throwable::class)
    override fun getValueSources(sig: ParameterSignature): List<PotentialAssignment> {
        val annotation: Between = sig.getAnnotation(Between::class.java)

        return (annotation.first..annotation.last)
                .map { PotentialAssignment.forValue("ints", it) }
    }
}
